### ISSUE TITLE
Failed to install vsphere-cloud-controller-manager

### ISSUE THREAD

**Issue by divyenpatel** [kubernetes/cloud-provider-vsphere#667]
### What happened?

```
# VERSION=1.25

# wget https://raw.githubusercontent.com/kubernetes/cloud-provider-vsphere/release-$VERSION/releases/v$VERSION/vsphere-cloud-controller-manager.yaml


# kubectl apply -f vsphere-cloud-controller-manager.yaml
serviceaccount/cloud-controller-manager created
secret/vsphere-cloud-secret created
configmap/vsphere-cloud-config created
rolebinding.rbac.authorization.k8s.io/servicecatalog.k8s.io:apiserver-authentication-reader created
clusterrolebinding.rbac.authorization.k8s.io/system:cloud-controller-manager created
clusterrole.rbac.authorization.k8s.io/system:cloud-controller-manager created
Error from server (BadRequest): error when creating "vsphere-cloud-controller-manager.yaml": DaemonSet in version "v1" cannot be handled as a DaemonSet: strict decoding error: unknown field "spec.template.spec.nodeAffinity", unknown field "spec.template.spec.volumes[0].configMap.affinity"
```

### What did you expect to happen?

There should not be any error in the sample deployment file provided on the repository

### How can we reproduce it (as minimally and precisely as possible)?

download following file, change vSphere credentials and execute kubectl apply -f with this file.

https://raw.githubusercontent.com/kubernetes/cloud-provider-vsphere/release-1.25/releases/v1.25/vsphere-cloud-controller-manager.yaml

### Kubernetes version

<details>

```console
# kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"5835544ca568b757a8ecae5c153f317e5736700e", GitTreeState:"clean", BuildDate:"2022-09-21T14:33:49Z", GoVersion:"go1.19.1", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:49:09Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider or hardware configuration

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Kernel (e.g. `uname -a`)

<details>

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### Others

<details>

</details>

**Comment by divyenpatel** [comment]
The following section on the YAML is not aligned properly

```
            affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.kubernetes.io/control-plane
              operator: Exists
          - matchExpressions:
            - key: node-role.kubernetes.io/master
              operator: Exists
```

**Comment by peske** [comment]
As @divyenpatel noticed, section of the file is not indented properly. The proper indentation is:

```yaml
      volumes:
        - name: vsphere-config-volume
          configMap:
            name: vsphere-cloud-config
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: node-role.kubernetes.io/control-plane
                operator: Exists
            - matchExpressions:
              - key: node-role.kubernetes.io/master
                operator: Exists
```

(`affinity` has the same indentation as `volumes` above.)

**Comment by lubronzhan** [comment]
Fixed on main and release-1.25 branch.
Thanks for finding it. We need to improve our test

**Comment by lubronzhan** [comment]
https://github.com/kubernetes/cloud-provider-vsphere/issues/673
Created this one to fix it
